Sources/Datadog/URLSessionAutoInstrumentation/DDURLSessionDelegate.swift

```diff
@@ -13,9 +13,32 @@ import Foundation
 @objc
 open class DDURLSessionDelegate: NSObject, URLSessionTaskDelegate {
     var interceptor: URLSessionInterceptorType?
+    let firstPartyURLsFilter: FirstPartyURLsFilter?
 
     @objc
     override public init() {
+        firstPartyURLsFilter = nil
+        interceptor = URLSessionAutoInstrumentation.instance?.interceptor
+        if interceptor == nil {
+            let error = ProgrammerError(
+                description: """
+                `Datadog.initialize()` must be called before initializing the `DDURLSessionDelegate` and
+                first party hosts must be specified in `Datadog.Configuration`: `track(firstPartyHosts:)`
+                to enable network requests tracking.
+                """
+            )
+            consolePrint("\(error)")
+        }
+    }
+
+    // NOTE: RUMM-954 copy&pasting `init()` is a conscious decision.
+    // otherwise `DDURLSessionDelegateAsSuperclassTests` fails.
+    // if `init()` was made convenience and call the designated `init` below,
+    // that would result in potential breaking changes.
+    // host projects would need to change their `init()`s in subclasses.
+    // we can fix this in v2.0
```

Comment:
We cannot have this comment in public API. Instead it should be documenting the feature, especially:
* what is `firstPartyHosts` made for?
* how `firstPartyHosts` should be formatted (e.g. `example.com` not `https://example.com`)?
* what happens if `firstPartyHosts` are set for `DDURLSessionDelegate` and in global `Datadog.Configuration`?

Also, it would be cool to update `track(firstPartyHosts:)` documentation comment with a note that they can be overwritten per `DDURLSessionDelegate` instance.